addons/event/controllers/main.py

```diff
@@ -10,7 +10,8 @@ class EventController(Controller):
 
     @route(['''/event/<model("event.event"):event>/ics'''], type='http', auth="public")
     def event_ics_file(self, event, **kwargs):
-        files = event._get_ics_file()
+        frontend_lang = request.httprequest.cookies.get('frontend_lang')
```

Comment:
I was going to say that it's simple and LGTM but on second thoughts I'm wondering why this is not done automatically.

When you return a request.render response, the template content is automatically translated into the selected language.

Why is that different in this case? Shouldn't the "lang" be already set in the content? Can you investigate a bit further?

Thanks!

Reply:
Hello, 

Well the ics file is constructed by fetching some event fields so it's based on the user language and not on the language he chose to display on the frontend.

Reply:
Some more questions then.

- Can the cookie be empty? As event does not depend on website, not sure about language switcher.
- Should we respect the user language if you're not a public user?
- What about the event name on this line?
`('Content-Disposition', content_disposition('%s.ics' % event.name))`

Reply:
- The cookie can indeed be empty but it doesn't crash though :grinning: 
- Done
- Indeed ! I corrected that